gridstatus/miso_api.py

```diff
@@ -0,0 +1,321 @@
+            "MISO_API_PRICING_SUBSCRIPTION_KEY",
+        )
+        self.load_api_key = load_api_key or os.getenv("MISO_API_LOAD_SUBSCRIPTION_KEY")
+        self.default_timezone = "EST"
+
+    def get_lmp_day_ahead_hourly_ex_ante(self, date, end=None, verbose=False):
+        return self._get_pricing_data(
+            date,
+            end,
+            retrieval_func=self._get_lmp_day_ahead_hourly,
+            market=Markets.DAY_AHEAD_HOURLY,
+            version=EX_ANTE,
+            verbose=verbose,
+        )
+
+    def get_lmp_day_ahead_hourly_ex_post(self, date, end=None, verbose=False):
+        return self._get_pricing_data(
+            date,
+            end,
+            retrieval_func=self._get_lmp_day_ahead_hourly,
+            market=Markets.DAY_AHEAD_HOURLY_EX_POST,
+            version=EX_POST,
+            verbose=verbose,
+        )
+
+    def get_lmp_real_time_hourly_ex_post_prelim(self, date, end=None, verbose=False):
+        return self._get_pricing_data(
+            date,
+            end,
+            retrieval_func=self._get_lmp_real_time_hourly_ex_post,
+            market=Markets.REAL_TIME_HOURLY_PRELIM,
+            prelim_or_final=PRELIMINARY_STRING,
+            verbose=verbose,
+        )
+
+    def get_lmp_real_time_hourly_ex_post_final(self, date, end=None, verbose=False):
+        return self._get_pricing_data(
+            date,
+            end,
+            retrieval_func=self._get_lmp_real_time_hourly_ex_post,
+            market=Markets.REAL_TIME_HOURLY_FINAL,
+            prelim_or_final=FINAL_STRING,
+            verbose=verbose,
+        )
+
+    def get_lmp_real_time_5_min_ex_ante(self, date, end=None, verbose=False):
+        return self._get_pricing_data(
+            date,
+            end,
+            retrieval_func=self._get_lmp_real_time_5_min_ex_ante,
+            market=Markets.REAL_TIME_5_MIN_EX_ANTE,
+            verbose=verbose,
+        )
+
+    def get_lmp_real_time_5_min_ex_post_prelim(self, date, end=None, verbose=False):
+        return self._get_pricing_data(
+            date,
+            end,
+            retrieval_func=self._get_lmp_real_time_5_min_ex_post,
+            market=Markets.REAL_TIME_5_MIN_EX_POST_PRELIM,
+            prelim_or_final=PRELIMINARY_STRING,
+            verbose=verbose,
+        )
+
+    def get_lmp_real_time_5_min_ex_post_final(self, date, end=None, verbose=False):
+        return self._get_pricing_data(
+            date,
+            end,
+            retrieval_func=self._get_lmp_real_time_5_min_ex_post,
+            market=Markets.REAL_TIME_5_MIN_EX_POST_FINAL,
+            prelim_or_final=FINAL_STRING,
+            verbose=verbose,
+        )
+
+    # NOTE: this method does not use the support_date_range decorator. Instead
+    # it takes the output of a decorated function and processes that output all at once
+    # which is more efficient than processing each iteration of the decorator
```

Comment:
Hopefully, this is clear. We don't want to call `process_pricing_data` on each loop of the decorator function, so, we return the raw data lists instead then combine them into one list and do all the processing at once.